datastream-kafka/src/main/java/com/linkedin/datastream/kafka/KafkaTransportProvider.java

```diff
@@ -136,6 +140,40 @@ private int getSourcePartitionFromEvent(BrooklinEnvelope event) {
         event.getMetadata().getOrDefault(BrooklinEnvelopeMetadataConstants.SOURCE_PARTITION, "-1"));
   }
 
+  @Override
+  public void broadcast(String destinationUri, DatastreamProducerRecord record,
+      SendCallback onBroadcastComplete, SendCallback onEventComplete) {
+    Validate.isTrue(record.isBroadcastRecord(), "Trying to broadcast a non-broadcast type record.");
+
+    // Currently destination topic partition count will be queried from datastream destination metadata. This implies
+    // a restriction that datastream destination partition should be updated for broadcast to work correctly.
+    int partitionCount = _datastreamTask.getDatastreamDestination().getPartitions();
```

Comment:
For now we are doing with a limitation that for broadcast to work correctly we expect the partition count in datastream destination metadata to be correctly updated. 

I had some logic around dynamically querying partition count using AdminClient but Vaibhav mentioned that it will be expensive to do if broadcast is used by a client as frequently as send. Apparently for low level BMM we handle this by having a separate thread where we periodically check for partition count (at 60 sec interval). In future we can do that as an enhancement and live with the restriction for now. It should be fine for now since partition count is not a frequent operation.